### ISSUE TITLE
i18n support

### ISSUE THREAD

**Issue by cmbirk** [opengovfoundation/madison#513]
It looks like we have 2 good options for i18n, `angular-translate` and `angular-gettext`.  

There's a debate between the two creators on this GH issue:

 https://github.com/rubenv/angular-gettext/issues/3 

and a decent post on the subject here 

http://blog.trifork.com/2014/04/10/internationalization-with-angularjs/

@eduard-k and @savajr does this approach work for you guys as well?

**Comment by Vanuan** [comment]
As I see Madison is not a single page application, so I don't see advantage over Lavarel i8n

**Comment by cmbirk** [comment]
The 1.8 milestone includes a single page application structure for Madison.

**Comment by cmbirk** [comment]
Moving this to 2.0 milestone

**Comment by Vanuan** [comment]
Another option is built-in angular 1.4 support: https://github.com/angular/i18n

**Comment by barryvdh** [comment]
So, is there any update on this?

**Comment by krues8dr** [comment]
Actually, yes!  I got most of this figured out over on the editor in opengovfoundation/madison-editor@9f2f0803c91e87c38da12cca66cf7d2e2ef80f6b and I'm implementing this in Madison today.

**Comment by krues8dr** [comment]
Ok, I've made a big dent in this.  You should be able to create your own locale files and add them in the `public/locales` folder, matching the format of the `en_US.json` one I've created.  

There are a few outstanding pieces of this that I haven't had time to deal with, and I could certainly use some help on these:

* [ ] Add internationalization for emails #658
* [ ] Check internationalization for datepicker #659
* [ ] Add internationalization for serverside errors. #660
* [ ] Internationalization for database content. #661
* [ ] Internationalization for intro message. #662

There are, of course, probably other issues I haven't thought of or encountered yet as well.  I also need to write up documentation on the translations, but we're using [Angular Translate](http://angular-translate.github.io) for this if that helps.

**Comment by krues8dr** [comment]
This is all in the [`i18n`](https://github.com/opengovfoundation/madison/tree/i18n) branch at the moment, but will get merged into `master` very soon.

**Comment by krues8dr** [comment]
This is now in `master`, but this issue will remain open until those five items are resolved.  Also - if you do create a new locale file *please* contribute it back via a Pull Request to the main repo!  We really could use a *lot* of help on this front!

**Comment by krues8dr** [comment]
Updated this with new items.

**Comment by sethetter** [comment]
In v4, site is fully i18n'd, just lacking translations.
